### DISCUSSION TITLE
Error using `!relative` for base_path with 10.4

### DISCUSSION THREAD

**Discussion by Andre601** [facelessuser/pymdown-extensions#2236] [Q&A] [answered]
Updated to `10.4` to use the support for the `!relative` option of MkDocs in snippets, but every time I try to serve a live-preview is this error thrown:
```
$ mkdocs serve
ERROR   -  Config value 'markdown_extensions': Failed to load extension 'pymdownx.snippets'.
  File "C:\Users\Andreas\AppData\Local\Programs\Python\Python311\Lib\site-packages\mkdocs\utils\yaml.py", line 52, in __fspath__
    return os.path.join(self.value(), self.suffix)
                        ^^^^^^^^^^^^
  File "C:\Users\Andreas\AppData\Local\Programs\Python\Python311\Lib\site-packages\mkdocs\utils\yaml.py", line 98, in value
    raise exceptions.ConfigurationError(
ConfigurationError: The current file is not set for the '!relative' tag. It cannot be used in this context; the intended usage is within `markdown_extensions`.

Aborted with 1 configuration errors!
```

I doubt this is a configuration error on my end. All I did is change `base_path` from having a list of options, to having a single string option with value `!relative`

Here's the `markdown_extensions` config option with snippets in it:
```yaml
markdown_extensions:
  - attr_list
  - toc:
      permalink: true
  - meta
  - pymdownx.superfences
  - pymdownx.magiclink:
      repo_url_shortener: true
      repo_url_shorthand: true
      social_url_shorthand: true
  - pymdownx.emoji:
      emoji_index: !!python/name:material.extensions.emoji.twemoji
      emoji_generator: !!python/name:material.extensions.emoji.to_svg
  - pymdownx.highlight
  - pymdownx.snippets:
      base_path: !relative
  - pymdownx.blocks.admonition:
      types:
        - note
        - abstract
        - info
        - tip
        - success
        - question
        - warning
        - failure
        - danger
        - bug
        - example
        - quote
  - pymdownx.blocks.details:
  - pymdownx.blocks.tab:
      alternate_style: true
```

**Comment by oprypin** [comment, marked as answer]
So this is just something that I need to fix in MkDocs.

* https://github.com/mkdocs/mkdocs/pull/3466

MkDocs' validation couldn't handle the fact that the extension tries to immediately resolve the path inside `__init__`.

If we want to be super pragmatic about it and get the fix to users sooner, pymdownx could also make a change to avoid running into this bug.
https://github.com/facelessuser/pymdown-extensions/commit/131aac09e67ecfce5243eb85c3afbe1e9e047f0d
-actually this particular `abspath` call doesn't even have to be moved anywhere else because there's already such a call in the one place where it matters.

Though of course this *would* make a difference if some other code changes the current working directory somewhere in the middle. So maybe not safe to remove that `abspath` after all :confused:

**Reply by facelessuser** [reply]
Okay, unless a specific proposal for pymdownx is identified, I will leave the fix up to MkDocs as it doesn't seem there is a clear recommendation for us to ease this issue. Thanks for the info @oprypin!
